test/e2e/replica_set_operator_upgrade/replica_set_operator_upgrade_test.go

```diff
@@ -41,19 +41,76 @@ func TestReplicaSetOperatorUpgrade(t *testing.T) {
 	t.Run("Basic tests", mongodbtests.BasicFunctionality(&mdb, true))
 	t.Run("Keyfile authentication is configured", tester.HasKeyfileAuth(3))
 	t.Run("Test Basic Connectivity", tester.ConnectivitySucceeds())
-	t.Run("Test SRV Connectivity", tester.ConnectivitySucceeds(WithURI(mdb.MongoSRVURI("")), WithoutTls(), WithReplicaSet((mdb.Name))))
+	t.Run("Test SRV Connectivity", tester.ConnectivitySucceeds(WithURI(mdb.MongoSRVURI("")), WithoutTls(), WithReplicaSet(mdb.Name)))
 	t.Run("Test Basic Connectivity with generated connection string secret",
 		tester.ConnectivitySucceeds(WithURI(mongodbtests.GetConnectionStringForUser(mdb, scramUser))))
 	t.Run("Test SRV Connectivity with generated connection string secret",
 		tester.ConnectivitySucceeds(WithURI(mongodbtests.GetSrvConnectionStringForUser(mdb, scramUser))))
 	t.Run("Ensure Authentication", tester.EnsureAuthenticationIsConfigured(3))
 	t.Run("AutomationConfig has the correct version", mongodbtests.AutomationConfigVersionHasTheExpectedVersion(&mdb, 1))
 
-	// upgrade the uperator to master
+	// upgrade the operator to master
 	config := setup.LoadTestConfigFromEnv()
 	err = setup.DeployOperator(config, "mdb", false, false)
 	assert.NoError(t, err)
 
 	// Perform the basic tests
 	t.Run("Basic tests", mongodbtests.BasicFunctionality(&mdb, true))
 }
+
+// TestReplicaSetOperatorUpgradeFrom0_7_2 is intended to be run locally not in CI.
+// It simulates deploying cluster using community operator 0.7.2 and then upgrading it using newer version.
+func TestReplicaSetOperatorUpgradeFrom0_7_2(t *testing.T) {
```

Comment:
As a followup to this I would make this test to be running on CI. But this needs some more work, as it would require installing CRDs for 0.7.2 and then master inside the test, which afaik we do not currently do.